geode-core/src/main/java/org/apache/geode/DataSerializer.java

```diff
@@ -2906,12 +2906,12 @@ public static void writeObject(Object o, DataOutput out, boolean allowJavaSerial
       return;
     }
 
-    DISALLOW_JAVA_SERIALIZATION.set(Boolean.TRUE);
+    disallowJavaSerializationForThread.set(Boolean.TRUE);
     try {
       writeObject(o, out);
     } finally {
-      DISALLOW_JAVA_SERIALIZATION.set(Boolean.FALSE); // with JDK 1.5 this can be changed to
-                                                      // remove()
+      disallowJavaSerializationForThread.set(Boolean.FALSE); // with JDK 1.5 this can be changed to
```

Comment:
We should start changing our code to ensure that ThreadLocals are removed from Threads:
```java
threadLocal.set(value);
try {
  // do some work
} finally {
  threadLocal.remove();
}
```

Reply:
@kirklund What's the advantage of `remove()` rather than setting it back to the default?